### PR TITLE
Issue 517: port shared-resource-manager utils from distributredlog

### PR DESCRIPTION
**Pull request by zhaijack**
Descriptions of the changes in this PR:

Try to portshared-resource-manager utils  from DistributedLog, so it can be used later on.

> ---
> Be sure to do all of the following to help us incorporate your contribution
> quickly and easily:
> 
> - [x] Make sure the PR title is formatted like:
>     `<Issue # or BOOKKEEPER-#>: Description of pull request`
>     `e.g. Issue 123: Description ...`
>     `e.g. BOOKKEEPER-1234: Description ...`
> - [x] Make sure tests pass via `mvn clean apache-rat:check install findbugs:check`.
> - [x] Replace `<Issue # or BOOKKEEPER-#>` in the title with the actual Issue/JIRA number.
> 
> ---

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/bookkeeper-precommit-pullrequest-docker/33/

File: bookkeeper-common/src/main/java/org/apache/bookkeeper/common/util/SharedResourceManager.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+    private static final SharedResourceManager SHARED = create();
+
+    public static SharedResourceManager shared() {
+        return SHARED;
+    }
+
+    public static SharedResourceManager create() {
+        return create(() -> Executors.newSingleThreadScheduledExecutor(
+            ExecutorUtils.getThreadFactory("zstream-shared-destroyer-%d", true)));
```
Comment (eolivelli) [inline bookkeeper-common/src/main/java/org/apache/bookkeeper/common/util/SharedResourceManager.java]: bkstream ?

Comment (zhaijack) [reply bookkeeper-common/src/main/java/org/apache/bookkeeper/common/util/SharedResourceManager.java]: Thanks for it. will change it.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/bookkeeper-precommit-pullrequest-docker/34/

**Pull request closed without merging**